core/src/main/scala/Json.scala

```diff
@@ -245,8 +230,90 @@ object Json {
    * TIM: this seems really hacky.
    */
   implicit lazy val GithubEventDecoder: DecodeJson[Github.Event] =
-    GithubReleaseEventDecoder.asInstanceOf[DecodeJson[Github.ReleaseEvent]] |||
-    GithubPingEventDecoder.asInstanceOf[DecodeJson[Github.Event]]
+    ((GithubDeploymentEventDecoder |||
+     GithubReleaseEventDecoder: DecodeJson[Github.Event]) |||
+     GithubPullRequestEventDecoder: DecodeJson[Github.Event]) |||
+     GithubPingEventDecoder
+
+  /*
+   * {
+   *   "deployment": {
+   *     "url": "https://api.github.com/repos/timperrett/example/deployments/107241174",
+   *     "id": 107241174,
+   *     "node_id": "MDEwOkRlcGxveW1lbnQxMDcyNDExNzQ=",
+   *     "sha": "fdb7da2ab3b2cd172e86c1af9adefa3523f6d65b",
+   *     "ref": "fdb7da2ab3b2cd172e86c1af9adefa3523f6d65b",
+   *     "task": "deploy",
+   *     "payload": "{ \"foo\": true }",
+   *     "original_environment": "production",
+   *     "environment": "production",
+   *     "description": null,
+   *     "creator": {
+   *      ....
+   *     },
+   *     "created_at": "2018-10-06T04:52:40Z",
+   *     "updated_at": "2018-10-06T04:52:40Z",
+   *     "statuses_url": "https://api.github.com/repos/timperrett/example/deployments/107241174/statuses",
+   *     "repository_url": "https://api.github.com/repos/timperrett/example"
+   *   },
+   *   "repository": {
+   *     "id": 140525376,
+   *     "node_id": "MDEwOlJlcG9zaXRvcnkxNDA1MjUzNzY=",
+   *     "name": "example",
+   *     "full_name": "timperrett/example",
+   *     ....
+   *   },
+   *   "sender": {
+   *    ....
+   *   }
+   * }
+  */
+  implicit val GithubDeploymentEventDecoder: DecodeJson[Github.Deployment] =
+    DecodeJson(z => for {
+      a <- (z --\ "deployment" --\ "id").as[Long]
+      x <- (z --\ "repository" --\ "full_name").as[String]
+      b <- Slug.fromString(x).map(DecodeResult.ok
+           ).valueOr(e => DecodeResult.fail(e.getMessage,z.history))
+      c <- (z --\ "deployment" --\ "ref").as[String]
+      d <- (z --\ "deployment" --\ "environment").as[String]
```

Comment:
Perhaps pull the cursor down into `deployment` into a variable so we're not constantly traversing the same key? Could do like:

```scala
for {
  deployment <- DecodeResult.ok(z --\ "deployment")
  a <- (deployment --\ "id").as[Long]
  ...
}
```